tests/net/trickle/src/main.c

```diff
@@ -206,103 +163,60 @@ static bool test_trickle_1_wait_long(void)
-		return false;
-	}
-
-	return true;
+	zassert_true(net_trickle_is_running(&t2), "Trickle 2 not running");
 }
 
-static bool test_trickle_1_stopped(void)
+static void test_trickle_1_stopped(void)
 {
-	if (net_trickle_is_running(&t1)) {
-		TC_ERROR("Trickle 1 running\n");
-		return false;
-	}
-
-	return true;
+	zassert_false(net_trickle_is_running(&t1), "Trickle 1 running");
 }
 
-static bool test_trickle_2_inc(void)
+static void test_trickle_2_inc(void)
 {
-	if (!net_trickle_is_running(&t2)) {
-		TC_ERROR("Trickle 2 is not running\n");
-		return false;
-	}
-
+	zassert_true(net_trickle_is_running(&t2), "Trickle 2 is not running");
 	token2++;
-
-	return true;
 }
 
-static bool test_trickle_1_update(void)
+static void test_trickle_1_update(void)
 {
-	if (!net_trickle_is_running(&t1)) {
-		TC_ERROR("Trickle 1 is not running\n");
-		return false;
-	}
+	zassert_true(net_trickle_is_running(&t1), "trickle 1 is not running");
 
 	token1 = token2;
-
-	return true;
 }
 
-static bool test_init(void)
+static void test_init(void)
 {
 	k_sem_init(&wait, 0, UINT_MAX);
-
-	return true;
 }
 
-static const struct {
-	const char *name;
-	bool (*func)(void);
-} tests[] = {
-	{ "trickle test init", test_init },
-	{ "trickle create", test_trickle_create },
-	{ "trickle start", test_trickle_start },
-	{ "trickle 1 check status", test_trickle_1_status },
-	{ "trickle 2 check status", test_trickle_2_status },
-	{ "trickle 1 wait timeout", test_trickle_1_wait },
-	{ "trickle 2 wait timeout", test_trickle_2_wait },
-	{ "trickle 1 update", test_trickle_1_update },
-	{ "trickle 1 check status", test_trickle_1_status },
-	{ "trickle 2 update", test_trickle_2_inc },
-	{ "trickle 2 check status", test_trickle_2_status },
-	{ "trickle 1 check status", test_trickle_1_status },
-
-#if CHECK_LONG_TIMEOUT > 0
-	{ "trickle 1 wait long timeout", test_trickle_1_wait_long },
-#endif
-	{ "trickle stop", test_trickle_stop },
-	{ "trickle 1 check stopped", test_trickle_1_stopped },
-};
-
-void main(void)
+/*test case main entry*/
+void test_main(void)
 {
-	int count, pass;
-
-	for (count = 0, pass = 0; count < ARRAY_SIZE(tests); count++) {
-		TC_START(tests[count].name);
-		test_failed = false;
-		if (!tests[count].func() || test_failed) {
-			TC_END(FAIL, "failed\n");
-		} else {
-			TC_END(PASS, "passed\n");
-			pass++;
-		}
-	}
-
-	TC_END_REPORT(((pass != ARRAY_SIZE(tests)) ? TC_FAIL : TC_PASS));
+	ztest_test_suite(test_tickle,
+			ztest_unit_test(test_init),
+			ztest_unit_test(test_trickle_create),
+			ztest_unit_test(test_trickle_start),
+			ztest_unit_test(test_trickle_1_status),
+			ztest_unit_test(test_trickle_2_status),
+			ztest_unit_test(test_trickle_1_wait),
+			ztest_unit_test(test_trickle_2_wait),
+			ztest_unit_test(test_trickle_1_update),
+			ztest_unit_test(test_trickle_1_status),
+			ztest_unit_test(test_trickle_2_inc),
+			ztest_unit_test(test_trickle_2_status),
+			ztest_unit_test(test_trickle_1_status),
+#if CHECK_LOG_TIMEOUT > 1
```

Comment:
This should be `#if CHECK_LONG_TIMEOUT > 0` like in the original code.

Reply:
@jukkar  Done.